src/libAtomVM/nifs.c

```diff
@@ -4171,6 +4174,9 @@ static term nif_code_load_abs(Context *ctx, int argc, term argv[])
         RAISE_ERROR(BADARG_ATOM);
     }
     char *path = malloc(strlen(abs) + strlen(".beam") + 1);
+    if (IS_NULL_PTR(path)) {
```

Comment:
In case of failure we need to `free(abs)` otherwise we leak it.

Reply:
Indeed, I missed that.

Fixed.